### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by sanaAyrml**
Helloo hiiii, merge meee @VahidZee

### CODE REVIEW

File: raster/lyft/data/module.py [hunk 1]
```diff
@@ -15,6 +15,19 @@
 DEFAULT_NUM_WORKERS = 4
 
 
+class Indexed_Dataset(Dataset):
```
Comment (vahidzee) [inline raster/lyft/data/module.py]: class names should follow PascalCase convention. change it to IndexedDataset

File: raster/lyft/data/module.py [hunk 2]
```diff
@@ -68,6 +87,17 @@ def __init__(
             'validation proportion should not be None'
         print('val\n\t*split:', self.val_split, '*batch_size:', self.val_batch_size, '*shuffle:', self.val_shuffle,
               '*num_workers:', self.val_num_workers, '*idxs:', val_idxs, '*proportion:', self.val_proportion)
+        # test
```
Comment (VahidZee) [inline raster/lyft/data/module.py:90]: This shouldn't always be executed, check if `stage == 'test'`.

File: raster/lyft/trainer.py [hunk 3]
```diff
@@ -131,36 +140,56 @@ def forward(self, inputs, targets: torch.Tensor, target_availabilities: torch.Te
...
-            res['loss'] = nll.mean() if not reg_pgd else (nll + self.hparams.pgd_reg_factor * adv_nll).mean()
+            loss = nll if not reg_pgd else (nll + self.hparams.pgd_reg_factor * adv_nll)
+        if return_trajectory:
+            res['loss'] = loss
+        else:
+            res['loss'] = loss.mean()
+        if return_trajectory:
+            res['pred'] = pred
+            res['conf'] = conf
+            return res
```
Comment (vahidzee) [inline raster/lyft/trainer.py]: this return is unnecessary, if `return_results` is true it'll happen anyways

File: raster/lyft/trainer.py [hunk 4]
```diff
@@ -131,36 +140,56 @@ def forward(self, inputs, targets: torch.Tensor, target_availabilities: torch.Te
...
 
     def step(self, batch, batch_idx, optimizer_idx=None, name='train'):
         is_val = name == 'val'
+        is_test = name == 'test'
         if self.global_step == 0:
             self.init_hparam_logs()
         result = self(batch['image'], batch['target_positions'], batch.get('target_availabilities', None),
-                      return_results=True, attack=not is_val)
+                      return_results=True, attack=not (is_val or is_test), return_trajectory=is_test)
+        # if not is_test:
```
Comment (vahidzee) [inline raster/lyft/trainer.py:190]: remove

**Pull request merged**